### PR TITLE
types: remove param `explicitTz` from `types.ParseTime`

### LINKED ISSUE #48573: remove param `explicitTz` from `types.ParseTime`
see description in #48574

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #48573

Problem Summary:

In #41146 a input parameter `explicitTz` was introduced to function `types.ParseTime` to avoid data race. Now after some refactoring works after `47355`, we use `types.Context` which is immutable to replace `stmtctx.StatementContext`. So it is still thread-safe after we remove `explicitTz`.

### What is changed and how it works?

Remove param `explicitTz` from `types.ParseTime` to keep a clear semantic for function `types.ParseTime` .

We also remove `explicitTz` in `Time.check` and now `Time.check` and `Time.Check` are all same. Then just remove `Time.check` to use `Time.Check` only

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

File: pkg/expression/helper.go [hunk 1]
```diff
@@ -89,8 +89,11 @@ func getTimeCurrentTimeStamp(ctx sessionctx.Context, tp byte, fsp int) (t types.
 // GetTimeValue gets the time value with type tp.
 func GetTimeValue(ctx sessionctx.Context, v interface{}, tp byte, fsp int, explicitTz *time.Location) (d types.Datum, err error) {
 	var value types.Time
+	tc := ctx.GetSessionVars().StmtCtx.TypeCtx()
+	if explicitTz != nil {
+		tc = tc.WithLocation(explicitTz)
```
Comment (lcwangchao) [inline pkg/expression/helper.go:94]: `tc.WithLocation(explicitTz)` will create a new context with old one not modified. So there is no data race any more

File: pkg/types/time.go [hunk 2]
```diff
@@ -1183,12 +1178,7 @@ func parseDatetime(ctx Context, str string, fsp int, isFloat bool, explicitTz *g
 	if overflow {
 		// Convert to Go time and add 1 second, to handle input like 2017-01-05 08:40:59.575601
 		var t1 gotime.Time
-		if explicitTz != nil {
-			t1, err = tmp.GoTime(explicitTz)
-		} else {
-			t1, err = tmp.GoTime(ctx.Location())
-		}
-		if err != nil {
+		if t1, err = tmp.GoTime(ctx.Location()); err != nil {
```
Comment (lcwangchao) [inline pkg/types/time.go:1181]: We only need to care about `ctx.Location()` here after `explicitTz` removed

File: pkg/types/time.go [hunk 3]
```diff
@@ -1220,11 +1210,7 @@ func parseDatetime(ctx Context, str string, fsp int, isFloat bool, explicitTz *g
 		if err != nil {
 			return ZeroDatetime, errors.Trace(err)
 		}
-		if explicitTz != nil {
-			t1 = t1.In(explicitTz)
-		} else {
-			t1 = t1.In(ctx.Location())
-		}
+		t1 = t1.In(ctx.Location())
```
Comment (lcwangchao) [inline pkg/types/time.go:1213]: We only need to care about `ctx.Location()` here after `explicitTz` removed

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**